src/libAtomVM/nifs.c

```diff
@@ -5119,6 +5125,99 @@ static term nif_unicode_characters_to_binary(Context *ctx, int argc, term argv[]
     return result_tuple;
 }
 
+static term nif_erlang_lists_subtract(Context *ctx, int argc, term argv[])
+{
+    UNUSED(argc)
+
+    term list1 = argv[0];
+    term list2 = argv[1];
+
+    if (!term_is_list(list1) || !term_is_list(list2)) {
+        return BADARG_ATOM;
```

Comment:
This will not raise the error, but it will return the atom `badarg`. You should raise the error.
Anyway there is a macro that does all the work for you: `VALIDATE_VALUE(list1, term_is_list);`